azure-configurationservice/tests/conftest.py

```diff
@@ -0,0 +1,113 @@
+import os
+import logging
+from msrestazure.azure_exceptions import CloudError
+from azure.configuration import AzureConfigurationClient
+from azure.configuration import ConfigurationSetting
+
+PAGE_SIZE = 100
+
+
+class AzConfigTestData:
+    def __init__(self):
+        self.key_uuid = None
+        self.label1 = None
+        self.label2 = None  # contains reserved chars *,\
+        self.label_uuid = None
+        self.label1_data = []
+        self.label2_data = []
+        self.no_label_data = []
+
+
+def _add_for_test(app_config_client, kv):
+    exist = bool(list(app_config_client.list_configuration_settings(keys=[kv.key], labels=[kv.label])))
+    if exist:
+        _delete_from_test(app_config_client, key=kv.key, label=kv.label)
+    return app_config_client.add_configuration_setting(kv)
+
+
+def _delete_from_test(app_config_client, key, label):
+    try:
+        app_config_client.delete_configuration_setting(key=key, label=label)
+    except CloudError:
+        logging.debug("Error occurred removing configuration setting %s %s" % (key, label))
+
+
+def _create_app_config_client():
+    connection_str = os.environ['AZ_CONFIG_CONNECTION']
+    return AzureConfigurationClient(connection_str)
+
+
+def setup_data():
+    test_data = AzConfigTestData()
+    app_config_client = _create_app_config_client()
+
+    key_uuid = "test_key_a6af8952-54a6-11e9-b600-2816a84d0309"
+    label_uuid = "1d7b2b28-549e-11e9-b51c-2816a84d0309"
+    label1 = "test_label1_" + label_uuid
+    label2 = "test_label2_*, \\" + label_uuid
+    test_content_type = "test_content_type"
+    test_value = "test_value"
+
+    test_data.key_uuid = key_uuid
+    test_data.label_uuid = label_uuid
+    test_data.label1 = label1
+    test_data.label2 = label2
+
+    test_data.label1_data.append(
```

Comment:
Is this really necessary? If I understand correctly how the tests are working, every time a test is going to run, it calls `setup_data` which creates all of this information.
When setting the data, you use `_add_for_test` which always queries the service for that key label, and if it is found it deletes it.
That implies way too much work for every test.

Is there a reason why you are doing this?

Reply:
For replay, I use fixed pre-generated uuid. So I delete and then recreate to make sure I have the expected data. If I don't delete and recreate, it might happen I get dirty data. People may manually change/lock/delete from the portal. Other test code may also pollute the data.

Reply:
Is there a way to do the replay in a different way where the key is not hardcoded for live test? this is avoiding the test to be able to run parallel.

@alzimmermsft  how are you guys doing the recording work vs live? how do you manage keys?

Reply:
There could be a way to parameterize the request and response. But I haven't yet known how to. I'll put this in the backlog. I think I can already run the test in parallel. I use a static uuid, which will hardly conflict with other generated uuid. There could be other issues, but not about the uuid.

@alzimmermsft Currently I use a pre-generated hardcode uuid for keys. Python's test framework records every live test. And we can switch to replay with one setting.

Reply:
Java does something similar where the framework records the live tests every time, on the switch to playback mode it uses keys stored in a variable set for each test's recording.